Add FAQ entry about login (#1450)
praw-dev/praw: docs/getting_started/faq.rst

you are trying to sign in with has 2FA enabled, and as such, either needs a 2FA
token or a refresh token to sign in. A refresh token is preferred, because then
you will not need to enter a 2FA token in order to sign in, and the session
will last for longer than an hour. Refer to :ref:`2FA` and :ref:`refresh_token`
in order to use the respective auth methods.

.. _faq3:

Q: Some options (like getting moderator logs from r/mod) keep on timing out.
How can I extend the timeout?

A: Set the timeout config option or initialize :class:`.Reddit` with a timeout
of your choosing.

.. _faq4:

> Q: Help, I keep on getting redirected to `/login`!

Reviewer: This would be a lot more searchable (which IMO is a goal for FAQs) if you include the exact exception text somewhere.

Author: There's a lot of things that can cause a redirect. Anything with Multireddit editing, trying to view upvoted, subreddit data, even as simple as trying to upvote

Reviewer: What I'm saying is that this section should include the text "Redirect: Redirect to /r/subreddit/login/ (You may be trying to perform a non-read-only action via a read-only instance.)" (which I got from [your comment](https://github.com/praw-dev/praw/issues/1448#issuecomment-620185217))

Author: So far, it seems that only subreddit moderation actions redirect to login.

Author: Maybe something like
```suggestion
Q: Help, I keep on getting redirected to ``/r/subreddit/login/``!
```

Reviewer: I don't care about what exactly the title of the question is. I want the full exception text to be on the page because searching for exception text is a very common debugging method.

If you wanted, it could be part of the answer, or perhaps an elaboration on the question in a new paragraph.

> A: PRAW is most likely on a read-only mode. This normally occurs when you log in

Reviewer: ```suggestion
A: PRAW is most likely in read-only mode. This normally occurs when you log in
```

> without a username/password or a refresh_token. You need to sign in with either

Reviewer: ```suggestion
without a username and password or a refresh token. You need to sign in with either
```

> one of those. See :ref:`oauth_options` to see the avaliable methods

Reviewer: I'd clarify this by saying something like "The exception indicates that you need to sign in to perform this action."

> A: PRAW is most likely in read-only mode. This normally occurs when you log in
> without a username and password or a refresh token. You need to sign in with either
> one of those. See :ref:`oauth_options` to see the avaliable methods
> to sign in. 

Author: Would something like this be better phrasing?

```suggestion
A: PRAW is most likely in read-only mode. This normally occurs when you authenticate
without a username and password or a refresh token. You need to authenticate with either
one of those. See :ref:`oauth_options` to see the available authentication methods.
```

Reviewer: My previous comment was motivated by the ambiguity of "you need to" — *why* do I need to, or to achieve *what*? So I'd like something explaining the purpose of signing in.

Author: In that case, would this be better:
```suggestion
A: PRAW is most likely in read-only mode. This normally occurs when PRAW is
authenticated without a username and password or a refresh token. In order to perform
this action, the Reddit instance needs to be authenticated. See :ref:`oauth_options` to
see the available authentication methods.
```

Reviewer: Yes, that's great.

A2: You could also be using a Reddit instance in read-only mode. To set the 

> instance to not be in read_only mode, do `reddit.read_only = False`.

Reviewer: Does this happen? I doubt that people authenticate with user/pass and then manually enable read only mode.

Author: I mean I don't see why not. It's very unlikely, but it is an alternate answer for a reason. There isn't a limit on questions nor answers.

Reviewer: I'd like to keep only "frequently useful answers" on the "frequently asked questions" page… 😛

I just think that if a PRAW user is confused about an exception they got, we should lead them down the path that solves their problem without confusing them with something that never happens, or happens very rarely.

Author: I see. To be fair, the attribute is probably well-documented.

> A2: You could also be using a Reddit instance in read-only mode. To set the 
> instance to not be in read_only mode, do `reddit.read_only = False`.
> 
> .. note:: Trying to set an instance of Reddit to non-read-only without being signed
>    in will raise a :class:`.ClientException`.

Author: ```suggestion
```
